spinalcordtoolbox/reports/qc2.py

```diff
@@ -343,6 +345,146 @@ def sct_deepseg(
+        path_qc=Path(path_qc),
+        command=command,
+        cmdline=list2cmdline(cmdline),
+        plane='Sagittal',
+        dataset=dataset,
+        subject=subject,
+    ) as imgs_to_generate:
+        # Load the spinal cord segmentation mask
+        im_sc = Image(fname_sc)
+        im_sc.change_orientation("RPI")
+        im_sc_data = im_sc.data
+
+        # Load the labeled lesion mask
+        im_lesion = Image(fname_label)
+        im_lesion.change_orientation("RPI")
+        im_lesion_data = im_lesion.data
+        label_lst = [label for label in np.unique(im_lesion.data) if label]
+
+        # Get the total number of lesions; this will represent the number of rows in the figure. For example, if we have
+        # 2 lesions, we will have two rows. One row per lesion.
+        num_of_lesions = len(label_lst)
+        # Get the sagittal lesion slices
+        sagittal_lesion_slices = np.unique(np.where(im_lesion_data)[0])
+        # Get the minimum sagittal slice with lesion. For example, if a lesion cover slices 7,8,9, get 7
+        min_sag_slice = min(sagittal_lesion_slices)
+        # Get the maximum sagittal slice with lesion. For example, if a lesion cover slices 7,8,9, get 9
+        max_sag_slice = max(sagittal_lesion_slices)
+        # Get the number of slices containing the lesion. For example, if a lesion cover slices 7,8,9, get 3
+        num_of_sag_slices = max_sag_slice - min_sag_slice + 1
+
+        # Get the midsagittal slice of the spinal cord
+        # Note: as the midsagittal slice is the same for all lesions, we can pick the first lesion ([0]) to get it
+        mid_sagittal_sc_slice = measure_pd.loc[0, 'midsagittal_spinal_cord_slice']
+
+        #  Create a figure
+        #  The figure has one row per lesion and one column per sagittal slice containing the lesion
+        fig, axes = mpl_plt.subplots(num_of_lesions,
+                                     num_of_sag_slices,
+                                     figsize=(num_of_sag_slices * 5, num_of_lesions * 5))
+
+        # Force axes to be a 2-dimensional array (to avoid indexing issues if we have only a single lesion or a single
+        # sagittal slice)
+        axes = np.atleast_2d(axes)
+
+        # Loop across lesions
+        for idx_row, lesion_label in enumerate(label_lst):
+            # NOTE: As the 'label_lesion()' function has been called at the beginning of the script, im_lesion_data is
+            # now "labeled" meaning that different lesions have different values, e.g., 1, 2, 3
+            # As we are looping across lesions, we get the lesion mask for the current lesion label
+            im_label_data_cur = im_lesion_data == lesion_label
+            # Restrict the lesion mask to the spinal cord mask (from anatomical level, it does not make sense to have
+            # lesion outside the spinal cord mask)
+            boolean_mask = (im_sc_data.data != 0)  # Nonzero -> True | Zero -> False; we use this in case of soft SC
```

Comment:
I'm pretty sure this is a typo:
```suggestion
            boolean_mask = (im_sc_data != 0)  # Nonzero -> True | Zero -> False; we use this in case of soft SC
```

I was surprised that the code didn't crash, but it turns out that numpy arrays happen to have a `data` attribute:
```py3
>>> from spinalcordtoolbox.image import Image
>>> im = Image('data/sct_example_data/t2/t2.nii.gz')
>>> type(im.data)
<class 'numpy.ndarray'>
>>> type(im.data.data)
<class 'memoryview'>
```

However trying to get a boolean mask from it gives a constant, not a mask:
```py3
>>> type(im.data != 0)
<class 'numpy.ndarray'>
>>> type(im.data.data != 0)
<class 'bool'>
```

Reply:
good catch! thanks!